### PR TITLE
I768 goals settings

### PR DESCRIPTION
**Pull request by Serhiy91**
review and merge after #781

### CODE REVIEW

File: app/imports/ui/react/organization-settings/components/KeyGoals/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  {
+    label: 'Number of completed & deleted goals',
+    valueKey: WorkspaceDefaultsTypes.DISPLAY_COMPLETED_DELETED_GOALS,
+  },
+];
+
+const KeyGoalsSettings = props => (
+  <CardBlockCollapse leftText="Key goals">
+    <CardBody className="card-block">
```
Comment (Dartv) [inline app/imports/ui/react/organization-settings/components/KeyGoals/index.js]: Use local `CardBlock` component instead

File: app/imports/ui/react/organization-settings/containers/KeyGoals/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  ]),
+  defaultProps({
+    [WorkspaceDefaultsTypes.DISPLAY_GOALS]:
+      WorkspaceDefaults[WorkspaceDefaultsTypes.DISPLAY_GOALS],
+    [WorkspaceDefaultsTypes.DISPLAY_COMPLETED_DELETED_GOALS]:
+      WorkspaceDefaults[WorkspaceDefaultsTypes.DISPLAY_COMPLETED_DELETED_GOALS],
+  }),
+  graphql(Mutation.UPDATE_ORGANIZATION_WORKSPACE_DEFAULTS, {
+    props: ({ mutate, ownProps: { _id } }) => ({
+      changeWorkspaceDefaults: (workspaceDefaults, callback) => mutate({
```
Comment (Dartv) [inline app/imports/ui/react/organization-settings/containers/KeyGoals/index.js]: Why callback? `changeWorkspaceDefaults` returns promise so you can use `catch` where you need it instead.

File: share/helpers.js [hunk 3]
```diff
@@ -255,3 +257,11 @@ export const removeRolesFromAllUsers = roles =>
   forEachOrgsUser(({ _id: organizationId }, { userId }) => {
     Roles.removeUsersFromRoles(userId, roles, organizationId);
   });
+
+export const getWorkplaceDefaultsUpdater = workspaceDefaults => {
```
Comment (Dartv) [inline share/helpers.js]: Workplace? Also, style violations:

File: share/helpers.js [hunk 4]
```diff
@@ -255,3 +257,11 @@ export const removeRolesFromAllUsers = roles =>
...
+  const setModifier = {};
+  forEachObjIndexed((value, key) => {
```
Comment (Dartv) [inline share/helpers.js]: Why not `mapObjIndexed`?

File: share/middleware/organization/checkOrgOwner.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Roles } from 'meteor/alanning:roles';
+import Errors from '../../errors';
+import { UserRoles } from '../../constants';
+
+export default (
+  getOrgId = (root, args) => args._id,
+) => (next, root, args, context) => {
+  const organizationId = getOrgId(root, args, context);
+
+  const canChange = Roles.userIsInRole(
```
Comment (Dartv) [inline share/middleware/organization/checkOrgOwner.js]: We already have a helper for this inside `/app/imports/api/checkers/roles.js`. Just move it to `/share/roles'.

File: share/middleware/organization/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export { default as checkOrgOwner } from './checkOrgOwner';
```
Comment (Dartv) [inline share/middleware/organization/index.js]: name it `ensureIsOrgOwner`

**Comment by Dartv** [comment]
You can also use `react-final-form` for this. It has a nice example of something very similar to what you're trying to accomplish. https://codesandbox.io/s/7k742qpo36

**Comment by Dartv** [comment]
Also, don't nest folders inside org-settings. That's the old and bad practice.

**Comment by Serhiy91** [comment]
@Dartv fixed

**Comment by Dartv** [comment]
peek 2018-03-05 10-48

**Pull request merged**